### ISSUE TITLE
web api odata - no support for entities/associations across schemas

### ISSUE THREAD

**Issue by johncrim** [Breeze/breeze.js#101]
In Edmx metadata, related entities can exist in separate schemas.   `parseCsdlNavProperty(entityType, csdlProperty, schema)` and `getAssociation(csdlNavProperty, schema)` currently do not handle the case where the association exists in a schema separate from the entitytype.  In the Microsoft.Data.Edm project, the default namespace (aka schema) for an association is the same namespace as the entitytype at the primary end of the association.  So for any association where the related entity types are in separate schemas, the current implementation will fail for at least one end of the association.

In addition, the error reporting for an association that can't be found is poor:
```
Unable to process returned metadata        Cannot read property 'end' of null

at http://localhost:55944/vendor/breeze/build/breeze.debug.js:16400:38
   at http://localhost:55944/vendor/datajs/datajs-1.1.2.js:1675:13
   at XMLHttpRequest.odata.defaultHttpClient.request.xhr.onreadystatechange (http://localhost:55944/vendor/datajs/datajs-1.1.2.js:2587:25)
```

`getAssociation(csdlNavProperty, schema)` is returning null because it can't find the association with the provided short name within the provided schema.  The call stack at this point is:

```
>	getAssociation [breeze.debug.js] Line 7322	Script
 	parseCsdlNavProperty [breeze.debug.js] Line 7180	Script
 	Anonymous function [breeze.debug.js] Line 7070	Script
 	completeParseCsdlEntityType [breeze.debug.js] Line 7069	Script
 	parseCsdlEntityType [breeze.debug.js] Line 7055	Script
 	Anonymous function [breeze.debug.js] Line 7013	Script
 	Anonymous function [breeze.debug.js] Line 7012	Script
 	parse [breeze.debug.js] Line 6984	Script
 	proto.importMetadata [breeze.debug.js] Line 6506	Script
 	Anonymous function [breeze.debug.js] Line 16401	Script
 	Anonymous function [datajs-1.1.2.js] Line 1675	Script
 	xhr.onreadystatechange [datajs-1.1.2.js] Line 2587	Script
 	[Async Call]	
 	odata.defaultHttpClient.request [datajs-1.1.2.js] Line 2608	Script
 	invokeRequest [datajs-1.1.2.js] Line 1655	Script
 	odata.request [datajs-1.1.2.js] Line 8206	Script
 	odata.read [datajs-1.1.2.js] Line 8167	Script
 	proto.fetchMetadata [breeze.debug.js] Line 16384	Script
 	proto.fetchMetadata [breeze.debug.js] Line 6653	Script
 	proto.fetchMetadata [breeze.debug.js] Line 13437	Script
 	proto.executeQuery [breeze.debug.js] Line 13527	Script
 	proto.execute [breeze.debug.js] Line 11613	Script
 	Core.Services.AdAiringDataAccessService.constructor [AdAiringDataAccess.ts] Line 76	TypeScript
```

Using the webApiOData data service adapter; using Breeze.Client 1.5.4.

I will attempt to provide a test case and a patch for this.

**Comment by wardbell** [comment]
Please be aware of the difference between Web API OData v.3 and Web API OData v.4. We don't support v.4 yet. We're working through a ton of issues with the MS OData team.

Take a look at [this stackoverflow post](http://stackoverflow.com/questions/30928916/one-to-many-navigation-property-not-working/30946713) if you want to scare yourself.

**Comment by johncrim** [comment]
Thanks for the response Ward.  I should have clarified, we're using OData v3 for this.  I'm aware that the $metadata changed completely for v4 and it's not yet supported in Breeze.

I implemented a fix in my branch that works in our use case.  The branch also includes a fix for #96, and is otherwise unchanged from master.

I'd like to add unit tests for the fixes for both #96 and #101 (good contributors always provide tests), but I didn't see any examples for testing metadata parsing, and I'd like to be consistent with the style and approach of the project.

Cheers.
John

**Comment by steveschmitt** [comment]
I incorporated your code changes in a separate commit.  Yes, the error reporting was very bad in that area.  Thanks a lot for your work on this.
